src/context/TostifyContext.tsx

```diff
@@ -1,7 +1,43 @@
-import React from 'react'
+import { ReactNode, createContext, useContext } from "react";
+import {ToastContextType} from '../interfaces/Auth'
+import { toast } from "react-toastify";
+import 'react-toastify/dist/ReactToastify.css';
+// Create a context with the defined type
+const ToastContext = createContext<ToastContextType | undefined>(undefined);
+
+// Custom hook to use the toast context
+export const useToast = (): ToastContextType => {
+  const context = useContext(ToastContext);
+  if (!context) {
+    throw new Error('useToast must be used within a ToastProvider');
+  }
+  return context;
+};
+// Toast provider component
+export const ToastProvider:  React.FC<{ children: ReactNode }>=({ children }) => {
```

Comment:
Consider adding the ToastContainer to the App or main component, and this will make the toast as a top-level component, and then you can call `toast.success` and `toast.error` in any descendant component.